src/measure_theory/integration.lean

```diff
@@ -392,6 +392,19 @@ theorem restrict_apply (f : α →ₛ β) {s : set α} (hs : is_measurable s) (a
   restrict f s a = if a ∈ s then f a else 0 :=
 by simp only [hs, coe_restrict]
 
+lemma restrict_const_eq_mul_indicator 
+    (f : α → ennreal) (s : set α) (x : ennreal) (A1 : is_measurable s) :
+    (f * ⇑((const α x).restrict s)) = λ ω, (x * (set.indicator s f ω)) := 
+begin
+  apply funext,
+  intro ω,
+  simp,
```

Comment:
`simp` should never be used in the "middle" of a goal. It should always complete whatever goal it's working on. The reason for this is, someone could add a new simp lemma at some point in the future that lets it do more here. Then your next line wouldn't work. These kinds of errors are very hard to debug.

You can use `simp only [...]` since adding new simp lemmas won't affect this. An easy way to make the change is to change `simp` to `squeeze_simp` and click on the output, which will replace the call with the corresponding `simp only`.

Reply:
This is a great piece of advice. A significant fraction of my code maintenance is because simp gets more powerful over time. "simp only" is introduced in:
https://leanprover.github.io/theorem_proving_in_lean/tactics.html
Do you think this rule could be rewritten as advice and placed there, or at least a mention of squeeze_simp?

Reply:
This is mentioned on our [simp tutorial](https://leanprover-community.github.io/extras/simp.html). I'm not sure if it's in scope for TPiL, but we should definitely add it to [our library style page](https://leanprover-community.github.io/contribute/style.html).